simulation/halsim_ds_socket/src/main/native/cpp/main.cpp

```diff
@@ -160,6 +168,15 @@ static void SetupEventLoop(wpi::uv::Loop& loop) {
   loop.SetData(loopData);
   SetupUdp(loop);
   SetupTcp(loop);
+  auto autoDisableTimer = Timer::Create(loop);
+  autoDisableTimer->timeout.connect([] {
+    uint64_t timeSinceLastDSPacket =
+        HAL_GetFPGATime(nullptr) - GetLastDSPacketTime();
+    if (timeSinceLastDSPacket > 100000) {
+      HALSIM_SetDriverStationEnabled(0);
+    }
+  });
+  autoDisableTimer->Start(Timer::Time(100), Timer::Time(100));
```

Comment:
Since the timer only runs every 100 ms, the worst case to disable is actually 200 ms (the check is run and the delta is 99.99 ms, and then we wait another 100 ms to check again).  Maybe this timer should run every 20 ms?